NuRadioReco/modules/sphericalWaveFitter.py

```diff
@@ -0,0 +1,173 @@
+                positions = self.__relative_positions[ich]
+                times = []
+                tmp = -1*get_time_delay_spherical_wave(pulser_position, ch_pair, n=n_index)
+                n_samples = -1*tmp * self.__sampling_rate
+                pos = int(len(self.__correlation[ich]) / 2 - n_samples)
+                corr += self.__correlation[ich, pos]
+                if debug_corr:
+                    ax = fig.add_subplot(len(self.__channel_pairs), 1, ich+1)
+                    ax.plot(self.__correlation[ich])
+                    ax.set_ylim((0, max(self.__correlation[ich])))
+                    ax.axvline(pos, label = 'reconstruction', lw = 1, color = 'orange')
+                    ax.axvline(self._pos_starting[ich], label = 'starting pos', lw = 1, color = 'green')
+                    ax.set_title("channel pair {}".format( ch_pair), fontsize = 5) 
+                    ax.legend(fontsize = 5)
+ 
+            if debug_corr:
+                fig.tight_layout()
+                fig.savefig("debug.pdf")
+
+            return -1*corr
+            
+            
+
+        trace = np.copy(station.get_channel(self.__channel_pairs[0][0]).get_trace())
+        self.__correlation = np.zeros((len(self.__channel_pairs), len(np.abs(scipy.signal.correlate(trace, trace))) ))
+
+        for ich, ch_pair in enumerate(self.__channel_pairs):
+            trace1 = np.copy(station.get_channel(self.__channel_pairs[ich][0]).get_trace())
+            trace2 =np.copy(station.get_channel(self.__channel_pairs[ich][1]).get_trace())
+       
+            t_max1 = station.get_channel(self.__channel_pairs[ich][0]).get_times()[np.argmax(np.abs(trace1))]
+            t_max2 = station.get_channel(self.__channel_pairs[ich][1]).get_times()[np.argmax(np.abs(trace2))]
+            corr_range = 50 * units.ns
+            snr1 = np.max(np.abs(station.get_channel(self.__channel_pairs[ich][0]).get_trace()))
+            snr2 = np.max(np.abs(station.get_channel(self.__channel_pairs[ich][1]).get_trace()))
+            if snr1 > snr2:
+                trace1[np.abs(station.get_channel(self.__channel_pairs[ich][0]).get_times() - t_max1) > corr_range] = 0
+            else:
+                trace2[np.abs(station.get_channel(self.__channel_pairs[ich][1]).get_times() - t_max2) > corr_range] = 0
+            self.__correlation[ich] = np.abs(scipy.signal.correlate(trace1, trace2))
+          
+
+
+        #### set positions for starting position ####
+        self._pos_starting = np.zeros(len(self.__channel_pairs))
+        for ich, ch_pair in enumerate(self.__channel_pairs):
+            positions = self.__relative_positions[ich]
+            times = []
+
+            tmp = get_time_delay_spherical_wave(start_pulser_position, ch_pair, n=n_index)
+            n_samples = tmp * self.__sampling_rate
+            self._pos_starting[ich] = int(len(self.__correlation[ich]) / 2 - n_samples)
+
+
+        method = 'Nelder-Mead'
```

Comment:
the line where this is used is commented. Needed? In the `if debug` it is redefined.